server/routerlicious/packages/services-core/src/mongoCheckpointRepository.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ICollection, ICheckpointRepository } from "./database";
+import { ICheckpoint, IDeliState, IScribe } from "./document";
+
+export class MongoCheckpointRepository implements ICheckpointRepository {
```

Comment:
Nice refactor here, nit, we can probably move the `localcheckpointEnable` and `documentRepository` logic down to here too, to make your code more Encapsulation